src/platform/backends/hyperv_api/virtdisk/virtdisk_snapshot.cpp

```diff
@@ -0,0 +1,284 @@
+            parent};
+
+    // The given child path must not exist
+    if (MP_FILEOPS.exists(child))
+        throw CreateVirtdiskSnapshotError{std::make_error_code(std::errc::file_exists),
+                                          "Child disk `{}` already exists",
+                                          child};
+
+    virtdisk::CreateVirtualDiskParameters params{};
+    params.predecessor = virtdisk::ParentPathParameters{parent};
+    params.path = child;
+    const auto result = VirtDisk().create_virtual_disk(params);
+    if (result)
+    {
+        mpl::debug(log_category, "Successfully created the child disk: `{}`", child);
+        return;
+    }
+
+    throw CreateVirtdiskSnapshotError{
+        result,
+        "Could not create the head differencing disk for the snapshot"};
+}
+
+void VirtDiskSnapshot::reparent_snapshot_disks(const VirtualMachine::SnapshotVista& snapshots,
+                                               const std::filesystem::path& new_parent) const
+{
+    mpl::debug(log_category,
+               "reparent_snapshot_disks() -> snapshots_count: {}, new_parent: {}",
+               snapshots.size(),
+               new_parent);
+
+    // The parent must already exist.
+    if (!MP_FILEOPS.exists(new_parent))
+        throw CreateVirtdiskSnapshotError{
+            std::make_error_code(std::errc::no_such_file_or_directory),
+            "Parent disk `{}` does not exist",
+            new_parent};
+
+    for (const auto& child : snapshots)
+    {
+        const auto& child_path = make_snapshot_path(*child);
+
+        if (!MP_FILEOPS.exists(child_path))
+            throw CreateVirtdiskSnapshotError{std::make_error_code(std::errc::file_exists),
+                                              "Child disk `{}` does not exists",
+                                              child_path};
+        if (const auto result = VirtDisk().reparent_virtual_disk(child_path, new_parent); !result)
+        {
+            mpl::warn(log_category,
+                      "Could not reparent `{}` to `{}`: {}",
+                      child_path,
+                      new_parent,
+                      static_cast<std::error_code>(result));
+            continue;
+        }
+        mpl::debug(log_category,
+                   "Successfully reparented the child disk `{}` to `{}`",
+                   child_path,
+                   new_parent);
+    }
+}
+
+void VirtDiskSnapshot::erase_impl()
+{
+
+    const auto& parent = get_parent();
+    const auto& self_path = make_snapshot_path(*this);
+    const auto& head_path = make_head_disk_path();
+    mpl::debug(log_category,
+               "erase_impl() -> parent: {}, self_path: {}",
+               parent ? parent->get_name() : "<none>",
+               self_path);
+
+    const auto& head_parent = [&]() {
+        std::vector<std::filesystem::path> chain{};
+        const auto list_r = VirtDisk().list_virtual_disk_chain(head_path, chain, 2);
+
+        if (chain.size() == 2)
```

Comment:
Could you please extract the 2 into a local constant? I was a little puzzled by this `if` until I realized the number is just the same that is passed above. A name would have helped.